services/src/main/java/org/exoplatform/news/NewsServiceImpl.java

```diff
@@ -1222,4 +1240,34 @@ public String formatMention(String text) {
     return sb.toString();
   }
 
+  /**
+   * Processes Mentioners who has been mentioned via the news body.
+   *
+   * @param body
+   */
+  private Set<String> processMentions(String body) {
+    Set<String> mentions = new HashSet<>();
+    mentions.addAll(parseMention(body));
+
+    return mentions;
+  }
+
+  private Set<String> parseMention(String str) {
+    if (str == null || str.length() == 0) {
+      return Collections.emptySet();
+    }
+
+    Set<String> mentions = new HashSet<>();
+    Matcher matcher = MentionInNewsNotificationPlugin.MENTION_PATTERN.matcher(str);
+    while (matcher.find()) {
+      String remoteId = matcher.group().substring(1);
+      Identity identity = loadUser(remoteId);
+      // if not the right mention then ignore
+      if (identity != null && !mentions.contains(identity.getId())) {
```

Comment:
Shouldn't it be identity.getRemoteId() instead of identity.getId()

Reply:
yesss but should we check the username `identity.getRemoteId()` or the full name `identity.getProfile().getFullName()` ?

Reply:
Yes, we need to add the userName
but Identity ID is an incremented number
and remoteId is the username

Reply:
okay, I see. We should change it [here too](https://github.com/Meeds-io/social/blob/develop/component/core/src/main/java/org/exoplatform/social/core/jpa/storage/RDBMSActivityStorageImpl.java#L740).

Reply:
It is diferent, in social we are storing the Identity ID in the activity itself. Here, we want to get the list of usernames of mentioned users to send them notifications.